docs/TURN.md

```diff
@@ -0,0 +1,142 @@
+
+- Nextcloud Talk will try direct P2P in the first place, use STUN if needed and TURN as last resort fallback. Thus to be most flexible and guarantee functionality of your Nextcloud Talk instance in all possible connection cases, you most properly want to setup a TURN server.
+
+## Install and setup _coTURN_ as TURN server
+
+#### 1. Download and install
+
+- On **Debian and Ubuntu** there are official repository packages available:
+    ```
+    sudo apt install coturn
+    ```
+- For many **other Linux derivatives and UNIX-likes** packages can be found on [https://pkgs.org/download/coturn](https://pkgs.org/download/coturn)
+- For all **other** cases check out the [Downloads in the wiki of coTURN](https://github.com/coturn/coturn/wiki/Downloads)
+
+
+#### 2. Make coturn run as daemon on startup
+
+- On **Debian and Ubuntu** you just need to enable the deployed sysvinit service by adjusting the related environment variable:
+    ```
+    sudo sed -i '/TURNSERVER_ENABLED/c\TURNSERVER_ENABLED=1' /etc/default/coturn
+    ```
+
+- Since **Debian Buster** and **Ubuntu disco** the package ships a systemd unit, which does not use `/etc/default/coturn` but is enabled automatically on install. To check whether a systemd unit is available:
+    ```
+    ls -l /lib/systemd/system/coturn.service
+    ```
+
+- If you installed coTURN manually, you may want to create an sysvinit service or systemd unit, or use another method to run the following during boot:
+    ```
+    /path/to/turnserver -c /path/to/turnserver.conf -o
+    ```
+
+- `-o` starts the server in daemon mode, `-c` defines the path to the config file.
+- There is also an official example available at [https://github.com/coturn/coturn/blob/master/examples/etc/coturn.service](https://github.com/coturn/coturn/blob/master/examples/etc/coturn.service)
+
+#### 3. Configure `turnserver.conf` for usage with Nextcloud Talk
+
+- Next you need to adjust the coTURN configuration file to work with Nextcloud Talk.
+- Choose the listening port (default is _3478_) and an authentication secret, where a random hex is recommended
+    ```
+    openssl rand -hex 32
+    ```
+
+- Then uncomment/edit the following settings accordingly:
+
+         listening-port=<yourChosenPortNumber>
+         fingerprint
+         lt-cred-mech # Only on coTURN below v4.5.0.8!
+         use-auth-secret
+         static-auth-secret=<yourChosen/GeneratedSecret>
+         realm=your.domain.org
+         total-quota=100
+         bps-capacity=0
+         stale-nonce
+         no-loopback-peers # Only on coTURN below v4.5.1.0!
+         no-multicast-peers
```

Comment:
The above part is formatted different now then in the (D)TLS section below. This leads to larger space between lines, see: https://nextcloud-talk.readthedocs.io/en/latest/TURN/#3-configure-turnserverconf-for-usage-with-nextcloud-talk